tensorpipe/channel/cuda_ipc/context_impl.cc

```diff
@@ -55,7 +52,45 @@ std::shared_ptr<CudaChannel> ContextImpl::createChannel(
 }
 
 bool ContextImpl::isViable() const {
-  return foundCudaLib_;
+  if (!foundCudaLib_) {
+    TP_VLOG(4) << "Channel context " << id_
+               << " is not viable because libcuda could not be loaded";
+    return false;
+  }
+
+  int deviceCount;
+  TP_CUDA_CHECK(cudaGetDeviceCount(&deviceCount));
+  for (int i = 0; i < deviceCount; ++i) {
+    cudaDeviceProp props;
+    TP_CUDA_CHECK(cudaGetDeviceProperties(&props, i));
+
+    if (!props.unifiedAddressing) {
```

Comment:
I remember we needed unified addressing, but I already forgot exactly why. Could you put a comment explaining it?

Reply:
According to the docs:
> IPC functionality is restricted to devices with support for unified addressing on Linux operating systems.

Reply:
Could you put this in the code, as a comment, so we'll remember it next time we read this?